build.gradle.kts

```diff
@@ -0,0 +1,274 @@
+ * limitations under the License.
+ */
+
+plugins {
+  idea
+  `java-platform`
+  signing
+  `maven-publish`
+  id("org.projectnessie.buildsupport.ide-integration")
+  id("org.projectnessie.nessie-project")
+  id("io.github.gradle-nexus.publish-plugin")
+}
+
+extra["maven.name"] = "Nessie"
+
+/*
+Main dependency handling happens in this build script.
+Declare versions as variables and add dependency constraints.
+This works for GitHub's dependabot, it can scan this file and create version-bump PRs.
+
+To use dependencies (without versions) from a "bom", import the bom using e.g.
+    implementation(platform("bom-coordinates"))
+    implementation("dependency-in-bom")
+
+The following declarations also contain some build dependencies.
+*/
+
+val versionAgroalPool = "1.16"
+val versionAgrona = "1.15.2"
+val versionAntlr = "4.10.1"
+val versionAssertJ = "3.23.1"
+val versionAwssdk = "2.17.207"
+val versionBouncyCastle = "1.70"
+val versionCel = "0.3.5"
+val versionCheckstyle = "10.3"
+// to fix circular dependencies with NessieClient, certain projects need to use the same Nessie
+// version as Iceberg/Delta has
+val versionClientNessie = "0.18.0"
+val versionDeltalake = "1.1.0-nessie"
+val versionDockerjava = "3.2.13"
+val versionErrorProneAnnotations = "2.12.1"
+val versionErrorProneCore = "2.13.1"
+val versionErrorProneSlf4j = "0.1.12"
+val versionGatling = "3.7.6"
+val versionGuava = "31.1-jre"
+val versionGoogleJavaFormat = "1.15.0"
+val versionGraalSvm = "22.0.0.2"
+val versionH2 = "2.1.210"
+val versionHadoop = "3.3.3"
+val versionIceberg = "0.13.1"
+val versionImmutables = "2.9.0"
+val versionJackson = "2.13.3"
+val versionJacksonSpark3 = "2.13.3"
+val versionJacoco = "0.8.8"
+val versionJakartaEnterpriseCdiApi = "2.0.2"
+val versionJakartaValidationApi = "2.0.2"
+val versionJandex = "2.4.2.Final"
+val versionJavaxServlet = "4.0.1"
+val versionJavaxWsRs = "2.1.1"
+val versionJaxrsApi21Spec = "2.0.1.Final"
+val versionJsr305 = "3.0.2"
+val versionJersey = "2.35"
+val versionJmh = "1.35"
+val versionJunit = "5.8.2"
+val versionLogback = "1.2.11"
+val versionMavenResolver = "1.7.3"
+val versionMaven = "3.8.5"
+val versionMockito = "4.6.1"
+val versionMongodbDriverSync = "4.3.4"
+val versionNessieApprunner = "0.21.4"
+val versionOpenapi = "3.0"
+val versionQuarkus = dependencyVersion("versionQuarkus") // ensure that plugin version is the same
+val versionQuarkusAmazon = "2.9.2.Final"
+val versionQuarkusLoggingSentry = "1.1.1"
+val versionParquet = "1.12.3"
+val versionPicocli = "4.6.3"
+val versionPostgres = "42.3.3"
+val versionProtobuf = "3.21.1"
+val versionReactor = "2020.0.19"
+val versionRestAssured = "4.5.1"
+val versionRocksDb = "7.2.2"
+val versionScala2_12 = "2.12.13"
+val versionScala2_13 = "2.13.8"
+val versionSlf4j = "1.7.36"
+val versionSpark31 = "3.1.2"
+val versionSpark32 = "3.2.1"
+val versionTestcontainers = "1.17.2"
+val versionWeld = "3.1.8.Final"
+
+extra["versionAntlr"] = versionAntlr
```

Comment:
Could we remove the version variables like `versionAntlr` and do ` extra["versionAntlr"] = "4.10.1"` directly?

Reply:
... and use `$dependencyVersion("versionAntlr")` when we need to reference it?

Reply:
That would break dependabot's pattern matching.

dependabot can resolve something like
```
val versionLibrary = "0.1.2"

dependencies {
  someConfigName("org.something:library:$versionLibrary")
}
```
but that's it.

I'd like to move to Gradle version catalogs (which are dumb toml files), but need https://github.com/dependabot/dependabot-core/issues/3121 for that